src/EnergyPlus/InputProcessing/CsvParser.cc

```diff
@@ -212,33 +222,71 @@ void CsvParser::parse_line(std::string_view csv, size_t &index, json &columns)
     size_t parsed_values = 0;
     const size_t num_columns = columns.size(); // Csv isn't empty, so we know it's at least 1
 
+    bool has_extra_columns = false;
+
     size_t this_cur_line_num = cur_line_num;
     size_t this_beginning_of_line_index = beginning_of_line_index;
 
+    auto getCurrentLine = [&]() {
+        size_t found_index = csv.find_first_of("\r\n", this_beginning_of_line_index);
+        std::string_view line;
+        if (found_index != std::string::npos) {
+            line = csv.substr(this_beginning_of_line_index, found_index - this_beginning_of_line_index);
+        }
+        return line;
+    };
+
     while (true) {
         token = look_ahead(csv, index);
         if (token == Token::LINE_END || token == Token::FILE_END) {
-            if (parsed_values != num_columns) {
+            if (has_extra_columns) {
+                warnings_.emplace_back(
+                    fmt::format("CsvParser - Line {} - Expected {} columns, got {}. Ignored extra columns. Error in following line.",
+                                this_cur_line_num,
+                                num_columns,
+                                parsed_values),
+                    false);
+                warnings_.emplace_back(getCurrentLine(), true);
+            } else if (parsed_values != num_columns) {
                 success = false;
 
-                size_t found_index = csv.find_first_of("\r\n", this_beginning_of_line_index);
-                std::string line;
-                if (found_index != std::string::npos) {
-                    line = csv.substr(this_beginning_of_line_index, found_index - this_beginning_of_line_index);
-                }
                 errors_.emplace_back(
                     fmt::format(
                         "CsvParser - Line {} - Expected {} columns, got {}. Error in following line.", this_cur_line_num, num_columns, parsed_values),
                     false);
-                errors_.emplace_back(line, true);
+                errors_.emplace_back(getCurrentLine(), true);
```

Comment:
When we reach the end of the line, we issue a warning if has_extra_columns, **otherwise** an error if the resulting number of parsed values is not the expected one.